dissect/target/plugins/os/unix/bsd/citrix/commandhistory.py

```diff
@@ -0,0 +1,143 @@
+import re
+from functools import cache
+from typing import Any, Iterator, List, Tuple, Union
+
+from dissect.target.helpers.fsutil import TargetPath
+from dissect.target.helpers.record import UnixUserRecord
+from dissect.target.helpers.utils import year_rollover_helper
+from dissect.target.plugin import export, internal
+from dissect.target.plugins.os.unix.history import (
+    CommandHistoryPlugin,
+    CommandHistoryRecord,
+)
+
+RE_CITRIX_NETSCALER_BASH_HISTORY_DATE = re.compile(r"(?P<date>[^<]+)\s")
+
+CITRIX_NETSCALER_BASH_HISTORY_RE = re.compile(
+    r"""
+        (?P<date>[^<]+)
+        \s
+        <
+            (?P<syslog_facility>[^\.]+)
+            \.
+            (?P<syslog_loglevel>[^>]+)
+        >
+        \s
+        (?P<hostname>[^\s]+)
+        \s
+        (?P<process_name>[^\[]+)
+            \[
+                (?P<process_id>\d+)
+            \]
+        :
+        \s
+        (?P<username>.*)\s
+        on\s
+            (?P<destination>[^\s]+)\s
+        shell_command=
+        \"
+            (?P<command>.*)
+        \"
+    $
+    """,
+    re.VERBOSE,
+)
+
+
+class CitrixCommandHistoryPlugin(CommandHistoryPlugin):
+    COMMAND_HISTORY_ABSOLUTE_PATHS = (("citrix-netscaler-bash", "/var/log/bash.log*"),)
+
+    def _find_history_files(self) -> List[Tuple[str, TargetPath, Any]]:
+        """
+        Find history files on the target that this plugin can parse.
+        """
+        history_files = []
+        for shell, history_absolute_path_glob in self.COMMAND_HISTORY_ABSOLUTE_PATHS:
+            for path in self.target.fs.glob(history_absolute_path_glob):
+                target_path = self.target.fs.path(path)
+                history_files.append((shell, target_path, None))
```

Comment:
As the `history_absolute_path_glob` is an absolute path, `path().glob()` raises an exception as it expects a relative glob rather than an absolute one. We could of course alter the values of the `COMMAND_HISTORY_ABSOLUTE_PATHS` from absolute to relative paths (e.g. `/var/log/bash.log*` to `var/log/bash.log*`) but in my opinion that is too confusing. Thus, I've kept it as-is for now. Let me know if you have an alternative solution in mind.

Reply:
Elsewhere we use `self.target.fs.path().glob(path.lstrip("/"))`